org.eclipse.lemminx/src/main/resources/META-INF/native-image/resource-config.json

```diff
@@ -0,0 +1,34 @@
+{
+  "resources":[
+    {"pattern":"\\QMETA-INF/services/javax.xml.parsers.SAXParserFactory\\E"},
+    {"pattern":"\\QMETA-INF/services/org.eclipse.lemminx.services.extensions.IXMLExtension\\E"},
+    {"pattern":"\\Qorg/apache/xml/resolver/etc/catalog.dtd\\E"},
+    {"pattern":"\\Qorg/apache/xml/resolver/etc/catalog.rng\\E"},
+    {"pattern":"\\Qorg/apache/xml/resolver/etc/catalog.xsd\\E"},
+    {"pattern":"\\Qorg/apache/xml/resolver/etc/xcatalog.dtd\\E"},
+    {"pattern":"\\Qorg/jsoup/nodes/entities-base.properties\\E"},
+    {"pattern":"\\Qorg/jsoup/nodes/entities-full.properties\\E"},
+    {"pattern":"\\Qschemas/xsd/XMLSchema.dtd\\E"},
+    {"pattern":"\\Qschemas/xsd/XMLSchema.xsd\\E"},
+    {"pattern":"\\Qschemas/xsd/datatypes.dtd\\E"},
+	{"pattern":"\\Qschemas/xsd/xml.xsd\\E"},
+	{"pattern":"\\Qorg/eclipse/lemminx/services/snippets/catalog-snippets.json\\E"},
```

Comment:
`pattern":"org/eclipse/lemminx/services/snippets/*$"` might work, see https://www.graalvm.org/reference-manual/native-image/Resources/

Might work for resources embedded in dependency jars too

Reply:
I modified it to use a regex for the snippet JSON files. What do you mean by using it for resources in dependency jars?